### ISSUE TITLE
Task 2 milestone 4: Improve performance of CNN

### ISSUE THREAD

**Issue by FLX-20** [FLX-20/Data-Science-Toolkits-and-Architectures-group-1#45]
Task 2
Instrument your code with Weights & Biases (within a Docker container). Choose an appropriate metric for
optimizing your ML Model. What is the reasoning behind it?
Note you can make a new Dockerfile (you don’t need to use docker-compose)
Your code should:
- Login to W&B (Tip: you can use ENTRYPOINT in a Dockerfile to run a shell script that logs you in (see
below))
- Train a Model
- Save and upload the trained model
- Log the value of the loss function (graphically)
- Log your metric (graphically), Tip use a Keras Metric
- (Optional) Log your current Git commit, also try to enforce that you have to first commit all changes before
W&B allows you to sync with the cloud otherwise you might log the wrong commit hash (and others do not
use the same code as you do when they want to reproduce the experiment by checking out the commit
hash). You can create a shell script for that (get Git commit hash and save as Environment Variable, use
Python code to retrieve the variable and save it in W&B)
- (Optional) Upload the output of the predict function as a file to W&B

Try to play with your Neural Network, by changing parameters or even its architecture. Make sure that you
log those changes automatically to W&B. Compare the runs on W&B, how do the metrics/loss change? It is
not necessary to “optimize” your model. Possible changes to the training:
- Architecture (add a Layer, delete a Layer, change hidden units)
- Different loss function
- Different optimizer
- Batch size, Learning rate, ...

**Comment by FLX-20** [comment]
As written in the issue of task 2 for milestone 3. I will take care of this issue now so that you can focus on task 2 of milestone 3 and improve your git skills.
